### PR TITLE
lsp: configure handler verbosity (don't spam nvim_err requiring enter to dismiss)

### PR DESCRIPTION
**Pull request by mjlbach**
* optionally allow handler to only report first instance of error

I tried a couple different methods, and this seemed the cleanest. Feedback welcome
Reported by @glacambre at https://github.com/neovim/nvim-lspconfig/issues/193

### CODE REVIEW

File: runtime/lua/vim/lsp/handlers.lua [hunk 1]
```diff
@@ -6,14 +6,28 @@ local api = vim.api
 local buf = require 'vim.lsp.buf'
 
 local M = {}
+local disabled_errors = {}
```
Comment (tjdevries) [inline runtime/lua/vim/lsp/handlers.lua:9]: should be configurable maybe?

Comment (mjlbach) [reply runtime/lua/vim/lsp/handlers.lua:9]: This is just what I'm using to cache disabled errors, but on that topic most of what has been done this far uses overriding handlers, do you have a preference/should we add a section in global options for dealing with language server settings?

**Comment by teto** [comment]
Reading your comments on Gitter, I feel like you wanted a way to filter :messages (when you mentioned "bypassing" it). It is huge work so I would be ok with an LSP solution first. Completely discarding the followup errors feel wrong: it makes debugging harder. I would be more comfortable with maybe tweaking their loglevel down and prefixing them with an "error already notified". I would like to see what we can do with https://github.com/neovim/neovim/pull/13843 since I find system-level notifications easier to work with: no need to press enter, I can ignore them more easily etc.

**Comment by mjlbach** [comment]
I'm not sure I'm completely understanding you, but I also agree that #13843 is a better solution, I mostly made this to address @glacambre's point in https://github.com/neovim/nvim-lspconfig/issues/193

**Pull request closed without merging**